### ISSUE TITLE
odd rendering of dataclass __init__ method

### ISSUE THREAD

**Issue by Lx** [mitmproxy/pdoc#404]
#### Problem Description
Given this `dataclass` definition:

```py
@dataclass(frozen=True, slots=True)
class ShiftChange:
    """A single instance of one shift being added or deleted, and when."""

    shift: Shift
    """The shift that this change concerns."""

    exists: bool
    """Whether the shift exists after the operation (i.e. it was added)."""

    window: TimeWindow
    """When the change occurred."""
```

`pdoc` renders the `__init__` method like this:

![image](https://user-images.githubusercontent.com/160164/172620206-fd400b2a-9539-465c-a4f2-0f560b69a03f.png)

I noted in particular:

* the comma placement between the `shift` and `exists` parameters
* the placement of the `window` parameter on the same line as the `exists` parameter, but not the `shift` parameter.

#### System Information

```text
❯ pdoc --version
pdoc: 12.0.1
Python: 3.10.4
Platform: macOS-12.4-arm64-arm-64bit
```

**Comment by mhils** [comment]
Could you please post a full self-contained example? The following works just fine for me:

`clerk.py`
```
from dataclasses import dataclass

class objects:
    class shift:
        class Shift: pass
    class timewindow:
        class TimeWindow: pass

Shift = objects.shift.Shift
TimeWindow = objects.timewindow.TimeWindow

@dataclass(frozen=True, slots=True)
class ShiftChange:
    """A single instance of one shift being added or deleted, and when."""

    shift: Shift
    """The shift that this change concerns."""

    exists: bool
    """Whether the shift exists after the operation (i.e. it was added)."""

    window: TimeWindow
    """When the change occurred."""
```


![image](https://user-images.githubusercontent.com/1019198/172624505-dc8a8547-af6d-4a77-8a5b-24af14623989.png)

**Comment by Lx** [comment]
If I put all the classes into one single file (in an attempt to offer you a self-contained example), then the problem disappears.  So this:


```py
from dataclasses import dataclass

# from .shift import Shift


class Shift:
    pass


class TimeWindow:
    pass


@dataclass(frozen=True, slots=True)
class ShiftChange:
    """A single instance of one shift being added or deleted, and when."""

    shift: Shift
    """The shift that this change concerns."""

    exists: bool
    """Whether the shift exists after the operation (i.e. it was added)."""

    window: TimeWindow
    """When the change occurred."""
```

renders with no issue:

![image](https://user-images.githubusercontent.com/160164/172631017-803c287c-fd05-4585-ba3c-f57d7e0c5800.png)

but if the `class Shift: pass` is commented out and the `from .shift import Shift` line is re-introduced, it looks like this again:

![image](https://user-images.githubusercontent.com/160164/172631240-a789d082-7924-4fb2-a104-406bc7428eeb.png)

However, notably, even if the sibling `shift.py` file contains only this:

```py
class Shift:
    pass
```

the formatting is still broken, so it's not a matter of what's in `shift.py`.

---

Also for what it's worth, it's apparently not limited to `__init__`—here's some odd rendering from another file:

![image](https://user-images.githubusercontent.com/160164/172629827-e8e52130-c5d4-424f-99eb-43d4f1f2d2d4.png)

noting in particular the splitting of my `old` parameter's type definition across different lines and indentation levels.

**Comment by mhils** [comment]
Ah, now I can repro. Thanks! Fixed in a34e22c3a939fed231dfaeca89082a24de6c97e8, a new release should be out in a few minutes. 😃

**Comment by Lx** [comment]
All looking perfect in v12.0.2.  Many thanks.
